### PR TITLE
[Merged by Bors] - feat(AlgebraicTopology/SimplicialSet): constructors for subfaces

### PR DESCRIPTION
**Pull request by jcommelin**
constructors for subfaces of the standard simplex and horn simplices

---

- [x] depends on: #9362

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicTopology/SimplicialSet.lean [hunk 1]
```diff
@@ -139,6 +178,104 @@ def hornInclusion (n : ℕ) (i : Fin (n + 1)) : Λ[n, i] ⟶ Δ[n] where
...
+/-- Alternative constructor for the edge of `Λ[n, i]` with endpoints `a` and `b`,
+assuming `3 ≤ n`. -/
+@[simps!]
+def edge₃ (n : ℕ) (i a b : Fin (n+1)) (hab : a ≤ b) (H : 3 ≤ n) :
+    Λ[n, i] _[1] :=
+  horn.edge n i a b hab <| Finset.card_le_three.trans H
+
+/-- The edge of `Λ[n, i]` with endpoints `j` and `j+1`.
+
+This constuctor assumes `0 < i < n`,
```
Comment (joelriou) [inline Mathlib/AlgebraicTopology/SimplicialSet.lean:234]:
```suggestion
This constructor assumes `0 < i < n`,
```

File: Mathlib/AlgebraicTopology/SimplicialSet.lean [hunk 2]
```diff
@@ -139,6 +178,104 @@ def hornInclusion (n : ℕ) (i : Fin (n + 1)) : Λ[n, i] ⟶ Δ[n] where
...
+  · simp only [← Fin.val_fin_le, Fin.coe_castSucc, Fin.val_succ, le_add_iff_nonneg_right, zero_le]
+  simp only [← Fin.val_fin_lt, Fin.val_zero, Fin.val_last] at h₀ hₙ
+  obtain rfl|hn : n = 2 ∨ 2 < n := by
+    rw [eq_comm, or_comm, ← le_iff_lt_or_eq]; omega
+  · revert i j; decide
+  · exact Finset.card_le_three.trans hn
+
+/-- The triangle in the standard simplex with vertices `k`, `k+1`, and `k+2`.
+
+This constuctor assumes `0 < i < n`,
```
Comment (joelriou) [inline Mathlib/AlgebraicTopology/SimplicialSet.lean:250]:
```suggestion
This constructor assumes `0 < i < n`,
```

**Review by joelriou — APPROVED** [review]

File: Mathlib/AlgebraicTopology/SimplicialSet.lean [hunk 3]
```diff
@@ -139,6 +178,104 @@ def hornInclusion (n : ℕ) (i : Fin (n + 1)) : Λ[n, i] ⟶ Δ[n] where
...
+
+open SimplexCategory Finset Opposite
+
+/-- The (degenerate) subsimplex of `Λ[n+2, i]` concentrated in vertex `k`. -/
+@[simps]
+def const (n : ℕ) (i k : Fin (n+3)) (m : SimplexCategoryᵒᵖ) : Λ[n+2, i].obj m := by
+  refine ⟨standardSimplex.const _ k _, ?_⟩
+  suffices ∃ x, ¬x = i ∧ ¬k = x by
+    simpa [← Set.univ_subset_iff, Set.subset_def, asOrderHom, not_or, Fin.forall_fin_one]
+  by_cases hi0 : i = 0
```
[the remark this answers is not in the document]
Comment (jcommelin) [reply Mathlib/AlgebraicTopology/SimplicialSet.lean]: Thanks! Great suggestion! Done

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**